nano/node/common.hpp

```diff
@@ -22,40 +21,57 @@ bool parse_tcp_endpoint (std::string const &, nano::tcp_endpoint &);
 
 namespace
 {
+uint64_t random_64_m{ 0 };
+uint64_t const & random_64 ()
+{
+	if (random_64_m == 0)
+	{
+		random_64_m = (uint64_t) nano::random_pool::generate_word32 (0, 0xffffffffUL) << 32 | nano::random_pool::generate_word32 (0, 0xffffffffUL);
+	}
+	return random_64_m;
+}
+
 uint64_t endpoint_hash_raw (nano::endpoint const & endpoint_a)
 {
 	assert (endpoint_a.address ().is_v6 ());
+	uint64_t result;
 	nano::uint128_union address;
 	address.bytes = endpoint_a.address ().to_v6 ().to_bytes ();
-	XXH64_state_t * const state = XXH64_createState ();
-	XXH64_reset (state, 0);
-	XXH64_update (state, address.bytes.data (), address.bytes.size ());
 	auto port (endpoint_a.port ());
-	XXH64_update (state, &port, sizeof (port));
-	auto result (XXH64_digest (state));
-	XXH64_freeState (state);
+	blake2b_state state;
+	blake2b_init (&state, sizeof (result));
```

Comment:
Let's use the blake2b_init_key initializer which does the key-initialization for us.

Reply:
It's much slower than blake2b_update, like 1.5-2 times